src/servers/http_server.cc

```diff
@@ -801,6 +801,23 @@ ValidateOutputParameter(triton::common::TritonJson::Value& io)
   return nullptr;  // success
 }
 
+TRITONSERVER_Error*
+CheckInputShape(
```

Comment:
You are checking only if the total number of elements is the same and not if each dimension is the same. Maybe rename to CheckInputElementCount or something that more clearly explains what this function is doing.